src/script/ismine.h

```diff
@@ -20,7 +20,9 @@ enum isminetype
     ISMINE_NO         = 0,
     ISMINE_WATCH_ONLY = 1 << 0,
     ISMINE_SPENDABLE  = 1 << 1,
+    ISMINE_USED       = 1 << 2,
     ISMINE_ALL        = ISMINE_WATCH_ONLY | ISMINE_SPENDABLE,
+    ISMINE_ALL_USED   = ISMINE_ALL | ISMINE_USED,
```

Comment:
Is it really necessary to add new ismine types? AFAICT, these are only used as ismine filters which I think could just as easily be done with a boolean. `IsMine()` isn't returning these values, and `ISMINE_ALL_USED` isn't used anywhere.

Reply:
You get most of the combinations with the new "used" flag, because a bunch of calls have a with and without "avoid reuse" state... so I think this is the cleanest approach.

`ISMINE_ALL_USED` is indeed not used; I added it to indicate that `ISMINE_ALL` is in fact not include the used ones. It's sort of a comment, with the added side effect that smart IDEs will show it when giving suggestions.